sdk/storage/azure-storage-blob/tests/test_blob_tags.py

```diff
@@ -258,71 +302,87 @@ def test_start_copy_from_url_with_tags(self, storage_account_name, storage_accou
         copy = copyblob.start_copy_from_url(sourceblob, tags=tags)
 
         # Assert
-        self.assertIsNotNone(copy)
-        self.assertEqual(copy['copy_status'], 'success')
-        self.assertFalse(isinstance(copy['copy_status'], Enum))
-        self.assertIsNotNone(copy['copy_id'])
+        assert copy is not None
+        assert copy['copy_status'] == 'success'
+        assert not isinstance(copy['copy_status'], Enum)
+        assert copy['copy_id'] is not None
 
         copy_content = copyblob.download_blob().readall()
-        self.assertEqual(copy_content, self.byte_data)
+        assert copy_content == self.byte_data
 
         resp = copyblob.get_blob_tags()
 
         # Assert
-        self.assertIsNotNone(resp)
-        self.assertEqual(len(resp), len(tags))
+        assert resp is not None
+        assert len(resp) == len(tags)
 
     @BlobPreparer()
-    def test_start_copy_from_url_with_tags_copy_tags(self, storage_account_name, storage_account_key):
+    @recorded_by_proxy
+    def test_start_copy_from_url_with_tags_copy_tags(self, **kwargs):
+        storage_account_name = kwargs.pop("storage_account_name")
+        storage_account_key = kwargs.pop("storage_account_key")
+        variables = kwargs.pop('variables', {})
+
         self._setup(storage_account_name, storage_account_key)
         tags = {"tag1": "firsttag", "tag2": "secondtag", "tag3": "thirdtag"}
         source_blob = self.bsc.get_blob_client(self.container_name, self._get_blob_reference())
         source_blob.upload_blob(b'Hello World', overwrite=True, tags=tags)
 
-        source_sas = generate_blob_sas(
+        expiry_time = self.get_datetime_variable(variables, 'expiry_time', datetime.utcnow() + timedelta(hours=1))
```

Comment:
For SAS, you shouldn't actually need to create a variable for dates. The `self.generate_sas` takes care of that for you somehow (not with variables but somehow)

Same with other SAS below and async SAS usage.